polars/polars-io/src/csv/read_impl/mod.rs

```diff
@@ -682,6 +702,67 @@ impl<'a> CoreReader<'a> {
                     })
                     .collect::<PolarsResult<Vec<_>>>()
             })?;
+            if let (Some(n_rows), Some(remaining_bytes)) = (self.n_rows, remaining_bytes) {
+                let rows_already_read = dfs.iter().map(|x| x.0.height()).sum::<usize>();
+                if rows_already_read < n_rows {
+                    dfs.push({
+                        let mut df = {
+                            let remaining_rows = n_rows - rows_already_read;
+                            // estimate the buffer size required for the remaining rows from the
+                            // rows already read (adding a 10% margin of error)
+                            let estimated_chunk_size =
```

Comment:
We don't need to estimate anymore as this last chunk will run single threaded. You can simply pass the `remaining_bytes` and ensure you pass `remaining_rows`, then the `parse_lines` should stop once that number of rows is read.